src/wrapper/Plumed.h

```diff
@@ -2874,28 +2955,258 @@ Plumed(Plumed&&p)__PLUMED_WRAPPER_CXX_NOEXCEPT :
+    shape_[i]=newindex;
+    shape_[i+1]=0;
+    return shape_;
+  }
+
+
+/// Helper functions for interpreting commands. **They are all internals**.
+/// cmd_helper is called when we have no shape information associated.
+/// cmd_helper_with_shape is called when we have shape information associated.
+/// The nocopy bool tells us if this pointer is pointing to a temporary variable,
+/// and thus PLUMED should not keep a copy.
+/// The variants below change for the type of the val argument
+
+/// cmd_helper with custom array val (includes std::array)
+  template<typename T, typename std::enable_if<wrapper::is_custom_array<T>::value, int>::type = 0>
+  void cmd_helper(const char*key,T&& val,bool nocopy=false) {
+    using value_type = typename wrapper::is_custom_array<T>::value_type;
+    constexpr std::size_t value_size=sizeof(value_type);
+    static_assert(value_size>0,"cannot use custom arrays of void types");
+    static_assert(sizeof(T)%value_size==0,"custom array has incorrect size");
+    std::size_t shape[] { sizeof(T)/sizeof(value_type), 0 };
+    cmd_helper_with_shape(key,reinterpret_cast<value_type*>(&val),shape,nocopy);
+  }
+
+/// cmd_helper with size/data val (typically, std::vector, std::string, small_vector, etc)
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value && wrapper::has_size_and_data<T>::value, int>::type = 0>
+  void cmd_helper(const char*key,T&& val,bool nocopy=false) {
+    std::size_t shape[] { wrapper::size(val), 0 };
+    cmd_helper_with_shape(key,val.data(),shape,nocopy);
+  }
+
+/// cmd_helper with raw pointer val
+/// In this case, the nocopy information is not propagated. We can indeed save the pointer, even if it's a temporary
+/// as it is in the case cmd("a",&a)
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value && !wrapper::has_size_and_data<T>::value && std::is_pointer<T>::value, int>::type = 0>
+  void cmd_helper(const char*key,T&& val,bool nocopy=false) {
+#if __PLUMED_WRAPPER_CXX_DETECT_SHAPES_STRICT
+    // this would be strict checking
+    // "a pointer without a specified size is meant to be pointing to a size 1 object"
+    std::size_t shape[] {  0, 0 };
+    if(val) shape[0]=1;
+    cmd_helper_with_shape(key,val,shape);
+#else
+    // for backward compatibility, the pointer is directly managed by SafePtr, with no size information
+    SafePtr s(val,0,nullptr);
+    cmd_priv(main,key,&s);
+#endif
+  }
+
+/// cmd_helper in remaining cases, that is when val is passed by value
+/// in this case, the nocopy information is not propagated, but the address of the data will be considered not copyable anyway
+/// by PLUMED ("pass by value")
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value && !wrapper::has_size_and_data<T>::value && !std::is_pointer<T>::value, int>::type = 0>
+  void cmd_helper(const char*key,T&& val,bool nocopy=false) {
```

Comment:
```suggestion
  void cmd_helper(const char*key,T&& val,bool =false) {
```
just to avoid an "unused argument/variable warning"

Reply:
Added this now: `(void) nocopy; // suppress warning`
I think that leaving the name makes the code clearer